Add failover docs (#3)
aterleto/jedis: docs/failover.md

# Failover with Jedis

Jedis supports failover for your Redis deployments. This is useful when:
1. You have more than one Redis deployment. This might include two independent Redis servers or two or more Redis databases replicated across multiple [active-active Redis Enterprise](https://docs.redis.com/latest/rs/databases/active-active/) clusters.
2. You want your application to connect to and use one deployment at a time.
3. You want your application to fail over to the next available deployment if the current deployment becomes unavailable.

Jedis will fail over to a subsequent Redis deployment after reaching a configurable failure threshold.
This failure threshold is implemented using a [circuit breaker pattern](https://en.wikipedia.org/wiki/Circuit_breaker_design_pattern).

You can also configure Jedis to retry failed calls to Redis.
Once a maximum number of retries have been exhausted, the circuit breaker will record a failure.

> When the circuit breaker reaches its failure threshold, a failover is triggered.

Reviewer: Its actually the next subsequent operation since now its forced open. So its not the operation that triggers it but the next

Author: See my suggested change.

> When the circuit breaker reaches its failure threshold, a failover is triggered.

Author: ```suggestion
When the circuit breaker reaches its failure threshold, a failover will be triggered on the subsequent operation.
```

The remainder of this guide describes:

* A basic failover configuration
* Supported retry and circuit breaker settings
* Failback and the cluster selection API

We recommend that you read this guide carefully and understand the configuration settings before enabling Jedis failover
in production.

## Basic usage

> To configure Jedis for failover, you specify an ordered list of Redis databases to connect to.

Reviewer: ```suggestion
To configure Jedis for failover, you specify an ordered list of Redis databases to connect.
```

Author: That is somewhat awkwardly phrased. I've pared it down even more. See my suggestion in the next review.

> To configure Jedis for failover, you specify an ordered list of Redis databases to connect to.

Author: ```suggestion
To configure Jedis for failover, you specify an ordered list of Redis databases.
```
